heron/api/src/java/org/apache/heron/api/metric/MultiCountMetric.java

```diff
@@ -38,7 +38,7 @@ public CountMetric scope(String key) {
 
   @Override
   public Map<String, Long> getValueAndReset() {
-    Map<String, Long> ret = new HashMap<>();
+    Map<String, Long> ret = new ConcurrentHashMap<>();
```

Comment:
ConcurrentHashMap might not be necessary in getValueAndReset(). This function is called every X seconds to sample the current metrics and export by Heron instance, so it is not called from multiple threads.